### PR TITLE
Extract `textSize` and `iconSize` from `Text` and `Icon` to modifiers

### PR DESCRIPTION
**Pull request by PavelHolec**

### CODE REVIEW

File: Sources/Orbit/Components/Button.swift [hunk 1]
```diff
@@ -30,10 +30,11 @@ public struct Button<LeadingIcon: View, TrailingIcon: View>: View {
                 trailingIcon
             }
         )
+        .textSize(custom: size.textSize.value)
```
Comment (sjavora) [inline Sources/Orbit/Components/Button.swift:33]: Can we put this inside OrbitButtonStyle?

File: Sources/Orbit/Components/Icon.swift [hunk 2]
```diff
@@ -411,17 +423,19 @@ struct IconPreviews: PreviewProvider {
...
 
-                + Icon(.informationCircle, size: .small)
+                + Icon(.informationCircle)
                     .iconColor(.blueNormal)
-                + Icon(.informationCircle, size: .small)
+                + Icon(.informationCircle)
                     .baselineOffset(.xxxSmall)
             )
+            .textSize(.small)
+            .iconSize(.small)
```
Comment (sjavora) [inline Sources/Orbit/Components/Icon.swift:438]: Isn't this something you said we don't need on slack?

File: Sources/Orbit/Support/Environment Keys/TextIsCopyableKey.swift [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import SwiftUI
+
+struct TextIsCopyableKey: EnvironmentKey {
+    static let defaultValue = false
+}
+
+public extension EnvironmentValues {
+
+    /// A text copyable status stored in a view’s environment, used for Orbit text components, such as `Text` or `Heading`.
+    var textIsCopyable: Bool {
```
Comment (sjavora) [inline Sources/Orbit/Support/Environment Keys/TextIsCopyableKey.swift:10]: I liked "selectable" better... is "copy" really the only option in all circumstances? Can't the text for example also be read by Siri or something?

File: Sources/Orbit/Components/Text.swift [hunk 4]
```diff
@@ -64,7 +67,7 @@ public struct Text: View, FormattedTextBuildable {
     }
 
     @ViewBuilder private var selectableLabelWrapper: some View {
-        if isSelectable {
+        if textIsCopyable {
```
Comment (sjavora) [inline Sources/Orbit/Components/Text.swift:70]: If you want to stick with copyable, you should probably rename the rest of "selectable" here.

**Review by sjavora — CHANGES_REQUESTED** [review]

File: Sources/Orbit/Support/Environment Keys/TextIsCopyableKey.swift (hunk 3 above)
Comment (PavelHolec) [reply Sources/Orbit/Support/Environment Keys/TextIsCopyableKey.swift:10]: Me too, but SwiftUI already has a `textSelection` modifier that "Use this method to indicate if text is selectable". But this is in regards to Mac selectability of any subtext. 

We are allowing text selection here, we only allow user to "long tap" and "copy" the text.

File: Sources/Orbit/Components/Text.swift (hunk 4 above)
Comment (PavelHolec) [reply Sources/Orbit/Components/Text.swift:70]: Updated

File: Sources/Orbit/Components/Icon.swift (hunk 2 above)
Comment (PavelHolec) [reply Sources/Orbit/Components/Icon.swift]: Updated

File: Sources/Orbit/Components/Button.swift (hunk 1 above)
Comment (PavelHolec) [reply Sources/Orbit/Components/Button.swift]: Updated

**Review by sjavora — APPROVED** [review]

**Pull request merged**